databricks/koalas/frame.py

```diff
@@ -1284,6 +1284,48 @@ def notnull(self):
 
     notna = notnull
 
+    def nunique(self, axis: int = 0, dropna: bool = True) -> 'ks.Series':
+        """
+        Return number of unique elements in the object.
+
+        Excludes NA values by default.
+
+        Parameters
+        ----------
+        axis : Can only be set to 0 at the moment.
+        dropna : Don’t include NaN in the count.
+
+        Returns
+        -------
+        The number of unique values as an int.
+
+        Examples
+        --------
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique()
+        A    3
+        B    1
+        Name: 0, dtype: int64
+
+        >>> ks.DataFrame({'A': [1, 2, 3], 'B': [np.nan, 3, np.nan]}).nunique(dropna=False)
+        A    3
+        B    2
+        Name: 0, dtype: int64
+        """
+        if axis != 0:
+            raise ValueError("The 'nunique' method only works with axis=0 at the moment")
+        if dropna:
+            res = self._sdf.select([F.countDistinct(Column(c))
+                                   .alias(c)
+                                    for c in self.columns])
+        else:
+            res = self._sdf.select([(F.countDistinct(Column(c))
+                                     + F.when(F.count(F.when(F.col(c).isNull(), 1).otherwise(None))
+                                              >= 1, 1).otherwise(0))
+                                   .alias(c)
+                                    for c in self.columns])
+        res = res.toPandas().T.iloc[:, 0]
+        return res
```

Comment:
Hm .. isn't it returning Pandas' ones, not Koalas series?

Reply:
It should probably return a Koalas Series, right?